packages/geo/src/lib/print/shared/print.service.ts

```diff
@@ -699,41 +611,47 @@ export class PrintService {
     map: IgoMap,
     resolution: number,
     size: Array<number>,
-    margins: Array<number>
+    margins: Array<number>,
+    legendPostion: PrintLegendPosition
   ) {
-    const status$ = new Subject();
 
     const mapSize = map.ol.getSize();
+    const viewResolution = map.ol.getView().getResolution();
     const extent = map.ol.getView().calculateExtent(mapSize);
-
     const widthPixels = Math.round((size[0] * resolution) / 25.4);
     const heightPixels = Math.round((size[1] * resolution) / 25.4);
+    const status$ = new Subject();
 
     let timeout;
+    map.ol.once('rendercomplete', async (event: any) => {
+      const mapCanvas = event.target.getViewport().getElementsByTagName('canvas')[0];
+      const mapResultCanvas = await this.drawMap(
+        [widthPixels, heightPixels],
+        mapCanvas
+      );
+
+      // Reset original map size
+      map.ol.setSize(mapSize);
+      map.ol.getView().setResolution(viewResolution);
+      this.renderMap(map, mapSize, extent);
+
+      await this.drawMapControls(map, mapResultCanvas, legendPostion);
 
-    map.ol.once('rendercomplete', (event: any) => {
-      const canvases = event.target.getViewport().getElementsByTagName('canvas');
       const mapStatus$$ = map.status$.subscribe((mapStatus: SubjectStatus) => {
         clearTimeout(timeout);
-
         if (mapStatus !== SubjectStatus.Done) {
           return;
         }
-
         mapStatus$$.unsubscribe();
-
         let status = SubjectStatus.Done;
         try {
-          for (const canvas of canvases) {
```

Comment:
Pourquoi avoir supprimer cette itération, dans quel cas de figure qu'on a plusieurs Canvas dans Openlayers? J'ai le feeling qu'on pourrait avoir une régression dans des cas spécifiques d'impression.

Reply:
ici nous n'avons qu'une seule canvas pour la carte c'est pourquoi j'ai supprimer la boucle.
@pelord pouvez-vous confirmer cela?